directus-remix/index.md

```diff
@@ -0,0 +1,331 @@
+```tsx
+import { LoaderFunctionArgs } from "@remix-run/node";
+import directus from "../lib/directus";
+import { readItem } from "@directus/sdk";
+import { useLoaderData } from "@remix-run/react";
+
+export const loader = async ({ params }: LoaderFunctionArgs) => {
+  const { slug } = params;
+
+  const page = await directus.request(readItem("pages", slug as string));
+  return page
+};
+
+export default function Page() {
+  const { title, content } = useLoaderData<typeof loader>();
+  return (
+    <div>
+      <h1>{title}</h1>
+      <div dangerouslySetInnerHTML={{ __html: content }}></div>
+    </div>
+  );
+}
+
+
+```
+
+Remix uses a special `useLoaderData()` hook to retrieve data from a `loader` function.
+
+Go to `http://localhost:3000/about`, replacing `about` with any of your item slugs. Using the Directus JavaScript SDK, the item with that slug is retrieved, and the page should show your data.
+
+```
+:::warning 404s and Trusted Content
+Non-existing slugs will result in a 404 error. Additionally,
+`dangerouslySetInnerHTML={{ __html: content }}` should only be used for trusted content.
+
+:::
+```
+
+## Creating Blog Posts With Directus
+
+Create a new collection called `authors` with a single text input field called `name`. Create one or more authors.
+
+Then, create a new collection called `posts` - make the Primary ID Field a "Manually Entered String" called   , which will correlate with the URL for the page. For example `hello-world` will later correlate to the page `localhost:3000/blog/hello-world`.
+
+Create the following fields in your `posts` data model:
+
+- a text input field called `title`
+- a WYSIWYG input field called `content`
+- an image relational field called `image`
+- a datetime selection field called `published_date` - set the type to 'date'
+- a many-to-one relational field called `author` with the related collection set to `authors`
+
+In Roles & Permissions, give the Public role read access to the `authors`, `posts`, and `directus_files` collections.
+
+Create some items in the posts collection - here's some sample data.
+
+### Create Blog Post Listing
+
+In Remix route file convention, `.` will create a `/` in the URL.
+Inside of the `app/routes` directory, create a new file called `blog._index.tsx` with the content:
+
+```tsx
+import directus from "~/lib/directus";
+import { readItems } from "@directus/sdk";
+import { useLoaderData } from "@remix-run/react";
+
+export const loader = async () => {
+  const posts = await directus.request(
+    readItems("posts", {
+      fields: ["slug", "title", "published_date", { author: ["name"] }],
+      sort: ["-published_date"],
+    })
+  );
+  return { posts };
+};
+
+
+
+```
+
+This query will retrieve the first 100 items (default), sorted by published date (descending order, which is latest first). It will only return the specific fields we request - `slug`, `title`, `published_date`, and the `name` from the related `author` item.
+
+Display the fetched data in HTML:
```

Comment:
It's not clear which file this code should be put in - make it clear it's at the bottom of the same file